src/authority.rs

```diff
@@ -773,7 +773,7 @@ pub enum Host<'host> {
     IPv4Address(Ipv4Addr),
 
     /// An IPv6 address. This will always be encased in brackets (`'['` and `']'`).
-    IPv6Address(Ipv6Addr),
+    IPv6Address(Ipv6Addr, Cow<'host, str>),
```

Comment:
Since the RFC says that a zone ID must be at least one character, I think it makes more sense to wrap this in an `Option`.